File: C++/Searching/Linear_search.cpp
// Linear search
// Time complexity = O(N)

#include <iostream>
using namespace std;

int linearSearch(int arr[], int size, int target)
{
    if (size == 0)
    {
        return -1;
    }
    for (int i = 0; i < size; i++)
    {
        if (arr[i] == target)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    int n, t;
    cout << "Enter number of elements in the array: " << endl;
    cin >> n;
    int arr[n];
    cout << "Enter the array: " << endl;
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    cout << "The array is: " << endl;
    for (int j = 0; j < n; j++)
    {
        cout << arr[j] << " ";
    }
    cout << endl;
    cout << "Enter the element you want to search: " << endl;
    cin >> t;
    cout << "The element is present at the index: " << linearSearch(arr, n, t);
    return 0;
}

File: Searching/src/com/company/LinearSearch.java
package com.company;

import java.util.Arrays;
import java.util.Scanner;

public class LinearSearch {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter an array of integers: ");
        int[] nums = new int[6];
        for (int j = 0; j < nums.length; j++) {
            nums[j] = input.nextInt();
        }
        System.out.println("The array of integers is: " + Arrays.toString(nums));
        System.out.println("Enter the target integer: ");
        int target = input.nextInt();
        int ans = linearsearch(nums, target);
        System.out.println("The index at which a given number is present is: " + ans);
    }

    static int linearsearch(int[] arr, int target) {
        if (arr.length == 0) {
            return -1;
        }
        for (int i = 0; i < arr.length; i++) {
            int element = arr[i];
            if (element == target) {
                return i;
            }
        }
        return -1;
    }
}

// Time complexity
// Best case = O(1)
// Worst case = O(N)